src/types/assets.t.ts

```diff
@@ -1,61 +1,89 @@
-import Web3 from "web3/types";
+import { ethers } from "ethers";
 
-export interface AssetGroupModel {
-  name: string;
-  AssetModel: AssetModel[];
+/// @notice Asset class interfaces
+
+export interface AssetClassConfig {
+  /** ethersProvider - ethers.js provider */
+  ethersProvider: any;
+  /** assets - Official contracts of the selected network */
+  assets: AssetsConfig;
+  /** assetIdentifier - The identifier of the asset contract to use */
+  assetIdentifier: string;
 }
 
-export interface AssetModel {
-  name: string;
-  cycle: string;
-  year: string;
-  collateral: string;
-  token: TokenModel;
-  emp: EmpModel;
-  pool: PoolModel;
-  apr?: AprModel;
+export interface EmpState {
+  expirationTimestamp: ethers.BigNumber;
+  collateralCurrency: string;
+  priceIdentifier: string;
+  tokenCurrency: string;
+  collateralRequirement: ethers.BigNumber;
+  disputeBondPercentage: ethers.BigNumber;
+  disputerDisputeRewardPercentage: ethers.BigNumber;
+  sponsorDisputeRewardPercentage: ethers.BigNumber;
+  minSponsorTokens: ethers.BigNumber;
+  timerAddress: string;
+  cumulativeFeeMultiplier: ethers.BigNumber;
+  rawTotalPositionCollateral: ethers.BigNumber;
+  totalTokensOutstanding: ethers.BigNumber;
+  liquidationLiveness: ethers.BigNumber;
+  withdrawalLiveness: ethers.BigNumber;
+  currentTime: ethers.BigNumber;
+  isExpired: boolean;
+  contractState: number;
+  finderAddress: string;
+  expiryPrice: ethers.BigNumber;
 }
 
-export interface TokenModel {
+/// @notice assets.json interfaces
+
+export interface PoolConfig {
+  /** address - Address of the pool contract */
   address: string;
-  decimals: number;
+  /** location - Location of the pool contract */
+  location: string;
 }
 
-export interface EmpModel {
+export interface EmpConfig {
+  /** address - Address of the emp contract */
   address: string;
+  /** new - Identifier to distinguish between the new and the old emp contracts */
   new: boolean;
+  /** type - Identifier to clarify the emp contract type */
+  type?: string;
 }
 
-export interface PoolModel {
+export interface TokenConfig {
+  /** address - Address of the token contract */
   address: string;
+  /** decimals - Decimals of the token contract */
+  decimals: number;
 }
 
-export interface AprModel {
-  force: number;
-  extra: number;
-}
-
-export interface Synths {
-  web3: Web3;
-  contracts: any;
-  addresses: any;
+export interface AssetConfig {
+  /** name - Name of the asset */
+  name: string;
+  /** cycle - Cycle of the asset */
+  cycle: string;
+  /** year - Year of the asset */
+  year: string;
+  /** collateral - Collateral of the asset */
+  collateral: string;
+  /** token - Token of the asset */
+  token: TokenConfig;
+  /** emp - Emp of the asset */
+  emp: EmpConfig;
+  /** pool - Pool of the asset */
+  pool: PoolConfig;
+  /** expired - Identifier for the expiry of the asset */
+  expired: boolean;
 }
```

Comment:
to keep comments where its really needed, instructive and not all over the place
- eventually everywhere and later after this pr, we will need to add it on the top of the object like that:
```js
/**
 * Asset specifications
 *
 * @member {string} name
 * @member {string} cycle
 * @member {string} year
 * @member {string} collateral
 * @member {TokenConfig} token
 * @member {EmpConfig} emp
 * @member {PoolConfig} pool
 * @member {boolean} expired Force expiration of the asset
 */
export interface AssetConfig {
```

```suggestion
/** AssetConfig - Asset specifications */
export interface AssetConfig {
  name: string;
  cycle: string;
  year: string;
  collateral: string;
  token: TokenConfig;
  emp: EmpConfig;
  pool: PoolConfig;
  expired: boolean; // Force expiration of the asset
}
```